Resources/public/js/recommendationrestclient.js

```diff
@@ -59,12 +60,29 @@
             }
         };
 
-        var attributes = '';
         for (var i = 0; i < this.attributes.length; i++) {
             attributes = attributes + '&attribute=' + this.attributes[i];
         }
 
-        xmlhttp.open('GET', this.endpointUrl + this.scenario + '.json' + '?numrecs=' + this.limit + '&' + 'contextitems=' + this.contextItems + attributes + '&contenttype=' + this.contentType + '&outputtype=' + this.outputType + '&categorypath=' + encodeURIComponent(this.categoryPath) + '&lang=' + this.language, true);
+        requestQueryString.push(
```

Comment:
additionally, it is common to do this only for get paramters, so you will typically have a more readable:
```javascript
var str = [
    'contextitems=' + this.contextItems,
    'contenttype=' + this.contentType
 ].join('&');
```